packages/contracts-bedrock/src/Safe/LivenessGuard.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard, GuardManager } from "safe-contracts/base/GuardManager.sol";
+import { ModuleManager } from "safe-contracts/base/ModuleManager.sol";
+import { SignatureDecoder } from "safe-contracts/common/SignatureDecoder.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+contract LivenessGuard is SignatureDecoder, BaseGuard {
+    /// @notice Emitted when a new set of signers is recorded.
+    /// @param signers An arrary of signer addresses.
+    event SignersRecorded(bytes32 indexed txHash, address[] signers);
+
+    Safe public immutable safe;
+    mapping(address => uint256) public lastSigned;
+
+    constructor(Safe _safe) {
+        safe = _safe;
+    }
+
+    /// @notice We just need to satisfy the BaseGuard interfae, but we don't actually need to use this method.
+    function checkAfterExecution(bytes32, bool) external pure {
+        return;
+    }
+
+    /// @notice Records the most recent time which any owner has signed a transaction.
+    function checkTransaction(
```

Comment:
If something happens where this method somehow always reverts, the safe is bricked right? I want to revisit the checks and external calls here to make sure that can't happen

Reply:
Resolving, this is tracked in the PR description.